Diagnostic/DistroSpecific.py

```diff
@@ -0,0 +1,319 @@
+        error, results = self.log_run_get_output(cmd)
+        return self.extract_om_path_and_version(results)
+
+    def extend_environment(self, env):
+        env.update({"SSL_CERT_DIR": "/etc/pki/tls/certs", "SSL_CERT_FILE": "/etc/pki/tls/cert.pem"})
+
+
+class Suse11Actions(RedhatActions):
+    def __init__(self, logger):
+        RedhatActions.__init__(self, logger)
+        self.certs_file = "/etc/ssl/certs/mdsd-ca-certs.pem"
+
+    def install_extra_packages(self, packages, with_timeout=False):
+        install_cmd = 'rpm -qi PACKAGE;  if [ ! $? == 0 ]; then zypper --non-interactive install PACKAGE;fi'
+        return self.log_run_multiple_cmds([install_cmd.replace("PACKAGE", p) for p in packages], with_timeout)
+
+    def install_required_packages(self):
+        return self.install_extra_packages(('rsyslog',), True)
+
+    # For SUSE11, we need to create a CA certs file for our statically linked OpenSSL 1.0 libs
+    def prepare_for_mdsd_install(self):
+        commands = (
+            r'cp /dev/null {0}'.format(self.certs_file),
+            r'chown 0:0 {0}'.format(self.certs_file),
+            r'chmod 0644 {0}'.format(self.certs_file),
+            r"cat /etc/ssl/certs/????????.[0-9a-f] | sed '/^#/d' >> {0}".format(self.certs_file)
+        )
+        return self.log_run_multiple_cmds(commands, False)
+
+    def extend_environment(self, env):
+        env.update({"SSL_CERT_FILE": self.certs_file})
+
+    def restart_rsyslog(self):
+        return self.log_run_ignore_output("""\
+if [ ! -f /etc/sysconfig/syslog.org_lad ]; then cp /etc/sysconfig/syslog /etc/sysconfig/syslog.org_lad; fi;
+sed -i 's/SYSLOG_DAEMON="syslog-ng"/SYSLOG_DAEMON="rsyslogd"/g' /etc/sysconfig/syslog;
+service syslog restart""")
+
+
+class Suse12Actions(RedhatActions):
+    def __init__(self, logger):
+        RedhatActions.__init__(self, logger)
+        self.certs_file = "/etc/ssl/certs/mdsd-ca-certs.pem"
+
+    def install_extra_packages(self, packages, with_timeout=False):
+        install_cmd = 'rpm -qi PACKAGE; if [ ! $? == 0 ]; then zypper --non-interactive install PACKAGE;fi'
+        return self.log_run_multiple_cmds([install_cmd.replace("PACKAGE", p) for p in packages], with_timeout)
+
+    def install_required_packages(self):
+        return self.install_extra_packages(('libgthread-2_0-0', 'ca-certificates-mozilla', 'rsyslog'), True)
+
+    def extend_environment(self, env):
+        env.update({"SSL_CERT_DIR": "/var/lib/ca-certificates/openssl", "SSL_CERT_FILE": "/etc/ssl/cert.pem"})
+
+
+class CentosActions(RedhatActions):
+    def __init__(self, logger):
+        RedhatActions.__init__(self, logger)
+
+    def install_extra_packages(self, packages, with_timeout=False):
+        install_cmd = 'rpm -qi PACKAGE; if [ ! $? == 0 ]; then yum install -y PACKAGE; fi'
+        return self.log_run_multiple_cmds([install_cmd.replace("PACKAGE", p) for p in packages], with_timeout)
+
+    def install_required_packages(self):
```

Comment:
Nor do I, but this code preserves the exact behavioral differences in the existing code base. We can puzzle out *why* those differences exist later. For now... no change.